### DISCUSSION TITLE
Melhores formas/práticas para criação de tabelas de dados

### DISCUSSION THREAD

**Discussion by felquis** [frontendbr/forum#741] [Perguntas e Respostas]
Fala pessoal, estou com um desafio muito massa de criar uma tabela de dados e essa tem um requisito clásico que é exibibir bastante dados, ser responsiva, acessível enfim toda a tarefa de casa bem feita, tem também questões de mobile vs desktop, referente a Layout/CSS/JavaScript, a forma de fazer fazer fetch dos dados e se tiver algum outro ponto também.

O projeto que estou usa React, porém experiencias em qualquer outra lib serão úteis, vi que existe algo tipo https://facebook.github.io/fixed-data-table/example-filter.html que achei legal como starting point.

Gostaria de saber de vocês qual foi as experiencias, erros e aprendizados que tiveram ao criar tabelas de dados para que eu possa aprender com vcs claro :dancer: :dagger: 

tables

**Comment by felipefialho** [comment]
@felquis Você se refere a layout/CSS?

**Comment by felquis** [comment]
@LFeh sim, e também aspectos que estão em volta como fazer fetch dos dados, os JSs, assuntos relacionados ao tipo dos dados que estão sendo exibidos etc.

**Comment by bernardodiasc** [comment]
Acho que essas caracteristicas sao possiveis sem muita dificuldade com CSS puro (e JS pro ajax loading). Uma coisa que é bastante desafiadora é fazer o "sticky header" da tabela.

**Comment by mumumilk** [comment]
Vi esse link faz um tempo, quando estava passando por esses questionamentos, e acredito que possa te ajudar.

https://uxdesign.cc/design-better-data-tables-4ecc99d23356

Na questao de responsividade, tu poderia mostrar uma coluna mais importante, e fazer ela expansivel para mostrar os outros dados em linhas.

**Comment by herberthenrique** [comment]
Sobre tabela responsiva, eu gosto bastante do jeito que é feito aqui:
https://codepen.io/AllThingsSmitty/pen/MyqmdM

Usando `data-attr`
No mobile você transforma a tabela em algo parecido com `cards`
Fica bem massa e é super simples :)
